src/lib/network_pool/transacation_pool_reorg_tests.ml

```diff
@@ -0,0 +1,433 @@
+          let sender = Command_spec.sender cmd in 
+          let content = Printf.sprintf !"%{sexp: Public_key.t} %{sexp: Command_spec.t}" test_keys.(sender.key_idx).public_key cmd in
+          `String content
+        ) |> Array.to_list in
+        
+        let log_minor = Array.map major_specs ~f:(fun cmd ->
+          let sender = Command_spec.sender cmd in 
+          let content = Printf.sprintf !"%{sexp: Public_key.t} %{sexp: Command_spec.t}" test_keys.(sender.key_idx).public_key cmd in
+          `String content
+        )|> Array.to_list in
+
+        let log_minor_accounts_state = Array.map minor_account_spec ~f:(fun spec ->
+          `String (Printf.sprintf !"%{sexp: Account_spec.t}\n" spec)
+        ) |> Array.to_list in
+        let log_major_accounts_state = Array.map major_account_spec ~f:(fun spec ->
+          `String ( Printf.sprintf !"%{sexp: Account_spec.t}\n" spec)
+        ) |> Array.to_list in
+        
+        [%log info] "Sequences" ~metadata:[
+          ("prefix" , `List log_prefix)
+          ; ("major" , `List log_major)
+          ; ("minor" , `List log_minor)
+          ; ("minor accounts state",`List log_minor_accounts_state )
+          ; ("major accounts state", `List log_major_accounts_state)
+        ];
+
+        let prefix = gen_commands_from_specs (Array.concat [prefix_specs]) test in
+        let minor = gen_commands_from_specs (Array.concat [minor_specs]) test in
+        let major = gen_commands_from_specs (Array.concat [major_specs]) test in
+        
+        let%bind () = advance_chain test (prefix @ major) in
+
+          Test.Resource_pool.handle_transition_frontier_diff_inner
+            ~new_commands:(List.map ~f:mk_with_status (prefix @ major))
+            ~removed_commands:(List.map ~f:mk_with_status (prefix @ minor))
+            ~best_tip_ledger:(Option.value_exn test.txn_pool.best_tip_ledger) test.txn_pool ;
+
+          Async.printf "Pool state: \n";
+          let pool_state = Test.Resource_pool.get_all test.txn_pool |> 
+           List.map ~f:(fun tx ->
+            let data = Transaction_hash.User_command_with_valid_signature.data tx in
+            let nonce = data |> User_command.forget_check
+             |> User_command.applicable_at_nonce |> Unsigned.UInt32.to_int in 
+             let fee_payer_pk = data |> User_command.forget_check
+             |> User_command.fee_payer |> Account_id.public_key
+             in
+             (fee_payer_pk,nonce)
+            ) in 
+
+          let log_pool_content = List.map pool_state ~f:(fun (fee_payer_pk,nonce) -> 
+            `String ( Printf.sprintf !"%{sexp: Public_key.Compressed.t} : %d" fee_payer_pk nonce )
+          ) in
+          
+          [%log info] "Pool state" ~metadata:[
+            ("pool state", `List log_pool_content )
+          ];
+
+          let assert_pool_contains pool_state (pk,nonce)= 
+            let actual_opt = List.find pool_state ~f:(fun (fee_payer_pk, actual_nonce ) -> 
```

Comment:
Rewrite this and next function: extract calculation of `actual_opt` into a helper function